src/javascript/core.js

```diff
@@ -6,69 +6,31 @@
 const Send = require('./core/send');
 const User = require('./core/user');
 const Session = require('./core/session');
-
-/**
- * Shared 'internal' scope.
- * @type {Object}
- */
 const settings = require('./core/settings');
 const utils = require('./utils');
 
-/**
- * Default properties for sending a tracking request.
- * @type {Object}
- * @return {Object} - The default settings for the component.
- */
-const defaultConfig = function () {
```

Comment:
I significantly refactored this core tracking class to simplify the logic.  I don't see an advantage of having some default config up here and some down in the `track` function.